### PR TITLE
Closes #81 | Create dataset loader for MTOP Intent Classification

### LINKED ISSUE #81: Create dataset loader for MTOP
Dataloader name: `mtop/mtop.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?mtop

|     Dataset| mtop |
|-------------|---|
| Description | An almost-parallel multilingual task-oriented semantic parsing dataset covering 6 languages and 11 domains. This is the first multilingual dataset which contains compositional representations that allow complex nested queries. |
| Subsets | Domain, Intent |
| Languages | tha |
| Tasks | Constituency Parsing, Semantic Role Labeling, Intent Classification |
| License | Unknown (unknown) |
| Homepage | https://huggingface.co/datasets/mteb/mtop_domain, https://huggingface.co/datasets/mteb/mtop_intent |
| HF URL | https://huggingface.co/datasets/mteb/mtop_domain, https://huggingface.co/datasets/mteb/mtop_intent |
| Paper URL | https://arxiv.org/abs/2008.09335 |

### PR DESCRIPTION
**Pull request by elyanah-aco**
Closes #81.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Comment by gentaiscool** [comment]
thank you for the PR. MTOP has multiple tasks (not only intent classification). Do you mind giving a more specific name for the dataset? (e.g., mtop_intent).

**Comment by sabilmakbar** [comment]
The dataset in question includes `mtop_domain` and `mtop_intent` subsets, which are the same task but with different labels of intent. The paper suggests that multiple tasks can be done, but the HF link for this dataset card only contains the `intent_classification` task. We might need to update the dataset card as the HF dataset doesn't seem to refer to the MTOP paper. 

In my opinion, the [HF Dataset](https://huggingface.co/datasets/tasksource/mtop) is more aligned with the original MTOP dataset, rather than the mentioned HF links that only create a derivative of it.

wdyt, @SamuelCahyawijaya @holylovenia @gentaiscool?

File: seacrowd/sea_datasets/mtop/mtop.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  booktitle   = {Proceedings of the 16th Conference of the European Chapter of the Association for Computational Linguistics: Main Volume},
+  publisher = {Association for Computational Linguistics},
+  year      = {2021},
+  url       = {https://aclanthology.org/2021.eacl-main.257},
+  doi       = {10.18653/v1/2021.eacl-main.257},
+  pages    = {2950-2962},
+}
+"""
+_LOCAL = False
+_LANGUAGE = "tha"
```
Comment (holylovenia) [inline seacrowd/sea_datasets/mtop/mtop.py:37]: `_LANGUAGES`

**Comment by holylovenia** [comment]
I think none of the HuggingFace URLs are provided by the original authors. The original authors merely provided a direct link to the data in the paper.

Is there any consistency problem(s) (e.g., different data sizes, different pairs, etc.) you found regarding the URLs that we use? @sabilmakbar @elyanah-aco 
https://huggingface.co/datasets/mteb/mtop_domain
https://huggingface.co/datasets/mteb/mtop_intent

If not, the dataloader as implemented should be fine if it works well. I'll continue reviewing this soon.

**Comment by sabilmakbar** [comment]
> - I think none of the HuggingFace URLs are provided by the original authors. The original authors merely provided a direct link to the data in the paper.

Yeah, but the latter HF link ([this one](https://huggingface.co/datasets/tasksource/mtop)) cites the MTOP paper and possibly has more variation of tasks than intent classification.

Perhaps the issue lies in these two datasets being a derivative dataset of MTOP instead of the original MTOP dataset, which I agree with @gentaiscool's suggestion to rename this dataset to MTOP Intent Classification.

However, I think the two HF links are quite aligned based on a few examples that I saw, and the `_intent` subset is more specific than the `_domain` subset on its label.

**Comment by holylovenia** [comment]
Going by [the datasheet](https://docs.google.com/spreadsheets/d/1ibbywsC1tQ_sLPX8bUAjC-vrTrUqZgZA46W_sxWw4Ss/edit?usp=sharing), the dataloader is supposed to cater for 2 subsets: `domain` and `intent`.

cc: @sabilmakbar @gentaiscool @elyanah-aco

**Comment by elyanah-aco** [comment]
> However, I think the two HF links are quite aligned based on a few examples that I saw, and the `_intent` subset is more specific than the `_domain` subset on its label.

Yes, this seems to be the case when I worked on this. Each `domain` can have multiple `intent` types, although the domain-intent mappings aren't available in the HF links.

**Comment by elyanah-aco** [comment]
@holylovenia Renamed dataset to `mtop_intent_classification` but for some reason, changing `mtop.py` to `mtop_intent_classification.py` gives `FileNotFoundError: No such file or directory` error when running `seacrowd_text` schema. It works for `source` schema though. Any possible reasons for this?

Also, do I need to add config to load all subsets as once?

**Comment by sabilmakbar** [comment]
> @holylovenia Renamed dataset to mtop_intent_classification but for some reason, changing mtop.py to mtop_intent_classification.py gives FileNotFoundError: No such file or directory error when running seacrowd_text schema. It works for source schema though. Any possible reasons for this?

For this, I can help debugging

> Also, do I need to add config to load all subsets as once?
Nope, it's okay as it is.

**Review by holylovenia — CHANGES_REQUESTED** [review]
Thanks for the changes, @elyanah-aco!! Tried and tested. The dataloader looks good to me. I'm not sure why you said it throws an error when you change the file name to `mtop_intent_classification.py`, because I tried using that file name and everything ran as it should. If you encounter a similar problem as @elyanah-aco, please let me know @sabilmakbar.

If there's no issue, could you please change the file name? Afterwards, I'll approve this PR and wait for @gentaiscool's review for the merge.

**Review by holylovenia — APPROVED** [review]
Thanks, LGTM!!

**Comment by sabilmakbar** [comment]
> @holylovenia Renamed dataset to mtop_intent_classification but for some reason, changing mtop.py to mtop_intent_classification.py gives FileNotFoundError: No such file or directory error when running seacrowd_text schema. It works for source schema though. Any possible reasons for this?

> Thanks for the changes, @elyanah-aco!! Tried and tested. The dataloader looks good to me. I'm not sure why you said it throws an error when you change the file name to mtop_intent_classification.py, because I tried using that file name and everything ran as it should. If you encounter a similar problem as @elyanah-aco, please let me know @sabilmakbar.

After testing this, it works too on my end. Perhaps you're on the wrong dir when executing the test suite or using the wrong args back then?

**Pull request merged**